functions/src/https/api/search/getSearchKeys.ts

```diff
@@ -0,0 +1,25 @@
+import * as functions from 'firebase-functions';
+
+import {
+  ALGOLIA_GENERALREQUESTS_INDEX,
+  ALGOLIA_UNAUTHENTICATEDREQUESTS_INDEX,
+  generateGeneralRequestsKey,
+  generateUnauthenticatedRequestsKey,
+} from '../../../algolia';
+
+export interface IgetSearchKeyReturn {
+  isAuthenticated: boolean;
+  searchKey: string;
+  indexName: string;
+}
+
+export const getSearchKey = functions.https.onCall(
+  (data, context): IgetSearchKeyReturn => {
```

Comment:
nit: I was thinking this would accept an `authenticated` key field in data indicating which index the client wants to access. If the user requests authenticated but uid is missing, throw new HttpsError('unauthenticated'). This makes it cleaner to handle the success and failure cases on the client. what you have here will still work.